SSA/Projects/RISCV64/Semantics2.lean

```diff
@@ -0,0 +1,642 @@
+/- we import these specific mathlib libraries to use Nat.cast_ofNat and neg_neg lemmas. -/
+import Mathlib.Order.Lattice
+import Mathlib.Data.Nat.Cast.Defs
+
+/-!
+  ## Dialect semantics bit-vectorization
+This file contains the `RV64` instruction semantics in a form that does
+not contain BitVec.toInt and BitVec.toNat. This file should replace
+the Semantics file in RISCV64 when complete.
+ToInt/ToNat calls hinder proof automation by bv_decide.
+To guarantee faithfullness to the semantics defined by the Sail-Lean RiscV64 model,
+we provide a proof with each rewrite that shows, that the toInt/ toNat free
+version is semantically equivalent to the defintion extracted from
+the Sail RiscV 64 modell in Lean.
+
+
+The instruction semantics suffixed by "bv" indicates that it is
+the purely bit vector defintion where we removed
+toInt/toNat calls.
+-/
+
+/-! ## Helper Bit Vector Lemmas  -/
+/- The lemmas below are helper lemmas used within proofs throughout the file. -/
+
+theorem sshiftRight_eq_sshiftRight_extractLsb {w : Nat}
+    {lw : Nat} (hll : 2^lw = w) (hlw : lw > 0 )
+    (x y : BitVec w)(h : BitVec.toNat y < 2^lw ) :
+        x.sshiftRight y.toNat = x.sshiftRight (y.extractLsb (lw - 1) 0).toNat := by
```

Comment:
```suggestion
theorem sshiftRight_eq_sshiftRight_extractLsb {w : Nat}
    {lw : Nat} {x y : BitVec w} (hll : 2^lw = w) (hlw : lw > 0)
    (h : BitVec.toNat y < 2^lw) :
        x.sshiftRight y.toNat = x.sshiftRight (y.extractLsb (lw - 1) 0).toNat := by
```
small formatting suggestion: we try to follow lean's guidelines and style, and in particular: 
* avoid spurious spaces
* call the "arguments" implicitly when possible (using the curly braces, unless this breaks the statement) and having first the implicit ones and then the hypotheses
could you check that this is the case for the remaining theorems as well? :)